src/guide/customization.rs

```diff
@@ -0,0 +1,398 @@
+//! - `Indenter`: a wrapper for formatters that inserts indentation after newlines for handling
+//! multi line error messages nicely
+//!
+//! Only one step remains, we need to override `eyre::Report`'s context parameter, which we will do
+//! with a type alias.
+//!
+//! ```rust
+//! pub(crate) type Report = eyre::Report<CustomContext>;
+//! ```
+//!
+//! And we're done! Here's what our final do nothing implementation of a custom context looks like:
+//!
+//! ```rust
+//! use eyre::Chain;
+//! use indenter::Indented;
+//!
+//! pub(crate) type Report = eyre::Report<CustomContext>;
+//!
+//! struct CustomContext;
+//!
+//! impl eyre::EyreContext for CustomContext {
+//!     fn default(_: &(dyn std::error::Error + 'static)) -> Self {
+//!         Self
+//!     }
+//!
+//!     fn debug(
+//!         &self,
+//!         error: &(dyn std::error::Error + 'static),
+//!         f: &mut core::fmt::Formatter<'_>,
+//!     ) -> core::fmt::Result {
+//!         if f.alternate() {
+//!             return core::fmt::Debug::fmt(error, f);
+//!         }
+//!
+//!         let errors = Chain::new(error).enumerate();
+//!
+//!         for (n, error) in errors {
+//!             writeln!(f)?;
+//!             write!(Indented::numbered(f, n), "{}", error)?;
+//!         }
+//!
+//!         Ok(())
+//!     }
+//! }
+//! ```
+//!
+//! # Lets Add Context to Our Context
+//!
+//! So now that we have the basic shape of a context and error report setup, let's go through the
+//! steps of adding a backtrace, by the end of this section we will have a context that almost
+//! identical to the `DefaultContext` provided by eyre and anyhow. The only difference between our
+//! implementation and the provided ones is that we won't jump through the extra hoops to
+//! conditionally disable backtraces when they're not available (e.g. we're on stable).
+//!
+//! To start we need to be able to store a backtrace in our context, lets modify our definition:
+//!
+//! ```rust
+//! use std::backtrace::Backtrace;
+//!
+//! struct CustomContext {
+//!     backtrace: Backtrace;
+//! }
+//! ```
+//!
+//! Next, lets modify the `default` fn to capture our backtrace when we first create our report
+//! from a `std::error::Error`:
+//!
+//! ```rust
+//! use std::backtrace::Backtrace;
+//!
+//! fn default(_: &(dyn std::error::Error + 'static)) -> Self {
+//!     Self {
+//!         backtrace: Backtrace::capture(),
+//!     }
+//! }
+//! ```
+//!
+//! Now we have a backtrace, but our report still wont print it, lets edit `debug to print this
+//! new member variable in our error report:
+//!
+//! ```rust
+//! use eyre::{Chain, Indenter};
+//! use std::backtrace::BacktraceStatus;
+//!
+//! fn debug(
+//!     &self,
+//!     error: &(dyn std::error::Error + 'static),
+//!     f: &mut core::fmt::Formatter<'_>,
+//! ) -> core::fmt::Result {
+//!     if f.alternate() {
+//!         return core::fmt::Debug::fmt(error, f);
+//!     }
```

Comment:
It'd be good to mention documentation which explains why `alternate` is being checked here.